### DISCUSSION TITLE
Subconditioning in JSON rules definition

### DISCUSSION THREAD

**Discussion by bhanu-affinidi** [hyperjumptech/grule-rule-engine#294] [Q&A]
Hi,
I have to create a rule where nested when conditions are required

`when { when c=d then return e} =true `

**Comment by niallnsec** [comment]
Have you tried writing this out as a single when statement with chained boolean operators?

For example:
```
when c = d && e = true
```

**Comment by bhanu-affinidi** [comment]
This is just a simple example my rule is very complex with multiple when conditions

**Reply by newm4n** [reply]
When you start to think that a rule could have a kind of "if" conditional flows, then you have not get the idea or "rule". Basicaly, if you can plan and design your rule properly, you won't need a conditional flow.
